lib/curlable.js

```diff
@@ -28,13 +30,26 @@ const generateRandomQueryId = () => {
   return s;
 };
 
-const Curlable = function(cmdline, inputLogger) {
-  this.logger = inputLogger || (() => {});
+const Curlable = function(cmdline, options, optionalLogger) {
+  this.options = options;
+  this.logger = optionalLogger || (() => {});
   this.spawnedProcess = child_process.spawn('/bin/bash', ['-c', cmdline]);
 
   this.queriesQueue = null;
 
   this.nextLineHandler = null;
+  this.ready = true;
+
+  if (options && options.prompt) {
+    const self = this;
```

Comment:
Arrow functions have this binding to parent context automatically, no need for closure.